### PR TITLE
Updates to terminology to match Northstar.

### LINKED ISSUE #54: Update post user endpoint to match northstar
use `term` and `id` as the [body content]( https://github.com/DoSomething/gladiator/blob/master/documentation/endpoints/users.md)

### PR DESCRIPTION
**Pull request by weerd**
#### What's this PR do?
This PR updates the terminology used for both the API endpoint for `/api/v1/users` and the User CRUD form. Both had to be updated since they both use the same form request for validation and thus the fileds need to be named the same.

Also updates the API documentation to reference the changes and update with an example response.

#### How should this be manually tested?
If User crud and POST request to `/api/v1/users` doesn't break, then all is good.

#### What are the relevant tickets?
Fixes #54 

---
@angaither @sbsmith86 @deadlybutter

### CODE REVIEW

File: app/Http/Transformers/UserTransformer.php [hunk 1]
```diff
@@ -23,6 +23,8 @@ public function transform(User $user)
             'mobile' => null,
             'signup' => null,
             'reportback' => null,
+            'created_at' => $user->created_at->toIso8601String(),
```
Comment (deadlybutter) [inline app/Http/Transformers/UserTransformer.php]: hm, why does this need to be exposed?

Comment (weerd) [inline app/Http/Transformers/UserTransformer.php]: Not sure, but we expose `created_at` and `updated_at` in Northstar so figured I'd follow suit.

Comment (deadlybutter) [inline app/Http/Transformers/UserTransformer.php]: ¯\_(ツ)_/¯

**Comment by deadlybutter** [comment]
:+1:

File: documentation/endpoints/users.md [hunk 2]
```diff
@@ -42,12 +42,28 @@ Content-Type: application/json
 curl -X POST \
   -H "Content-Type: application/json" \
   -H "Accept: application/json" \
-  -d '{"key":"clee@dosomething.org","type":"email","campaign_id":"362","campaign_run_id":"212"}' \
+  -d '{"key":"kallark@dosomething.org","type":"email","campaign_id":"362","campaign_run_id":"212"}' \
```
Comment (sbsmith86) [inline documentation/endpoints/users.md]: who dat

Comment (deadlybutter) [inline documentation/endpoints/users.md]: Gladiator! https://en.wikipedia.org/wiki/Gladiator_(Kallark)

**Comment by weerd** [comment]
tee hee :wink:

File: documentation/endpoints/users.md (hunk 2 above)
Comment (sbsmith86) [inline documentation/endpoints/users.md]: omg lol

**Comment by weerd** [comment]
image

Also in the login Email field placeholder :dancer:

**Comment by DFurnes** [comment]
That's such a good easter egg. :clap:

**Comment by sbsmith86** [comment]
amazing.

**Pull request merged**